internal/schema/resource.go

```diff
@@ -23,9 +23,11 @@ type Resource struct {
 	Tags           map[string]string
 }
 
-func CalculateCosts(resources []*Resource) {
-	for _, r := range resources {
-		r.CalculateCosts()
+func CalculateCosts(state *State) {
+	for _, resources := range [][]*Resource{state.ExistingState.Resources, state.PlannedState.Resources} {
```

Comment:
Maybe add a `.AllResources()` method to State.

Reply:
Makes total sense 👍🏼